src/lib/JIT.hs

```diff
@@ -47,15 +50,19 @@ import qualified RawName as R
 import Name
 import Imp
 import PPrint
-import Builder (TopBuilder (..), queryObjCache)
 import Logging
 import LLVMExec
 import Types.Primitives
-import Util (IsBool (..), bindM2)
+import Util (IsBool (..), bindM2, enumerate)
 
 -- === Compile monad ===
 
-type OperandSubstVal = SubstVal AtomNameC (LiftE Operand)
+data OperandSubstVal (c::C) (n::S) where
+  OperandSubstVal  :: L.Operand -> OperandSubstVal AtomNameC n
+  FunctionSubstVal :: L.Operand -> LLVMFunType -> IFunType -> OperandSubstVal ImpFunNameC   n
+  RenameOperandSubstVal :: Name c n -> OperandSubstVal c n
```

Comment:
We never emit those during the traversal, so perhaps we could call them top-level subst vals? or ffi subst vals?

Reply:
Added a comment explaining that that's the only way in which we currently use it